lib/delete-old-caches.js

```diff
@@ -9,11 +9,28 @@ self.addEventListener('activate', function(event) {
           return (cacheName.indexOf('$$$inactive$$$') === -1 && cacheName.indexOf(CACHE_PREFIX) === 0 && cacheName !== CACHE_VERSION);
         }).map(function(cacheName) {
           logDebug('Deleting out of date cache:', cacheName);
-          return caches.delete(cacheName);
+          return caches.delete(cacheName).then(function() {
```

Comment:
@bourquep why not check for existence of function here instead of catching non existence of function in `_postDeleteCacheHook`?  Seems like that would be simpler?  Something like
```js
if (brocswPostDeleteCacheHook) {
  return brocswPostDeleteCacheHook(cacheName);
} else {
  return Promise.resolve();
}
```